app/components/Form/ProjectAnnualReportForm.tsx

```diff
@@ -86,15 +96,24 @@ const ProjectAnnualReportForm: React.FC<Props> = (props) => {
     );
   }, [projectRevision.projectAnnualReportFormChanges]);
 
+  const overdue = isOverdue(
+    projectRevision.upcomingAnnualReportFormChange?.asReportingRequirement
+      .reportDueDate
+  );
+
+  const variant = determineVariant(
+    projectRevision.projectAnnualReportFormChanges,
+    overdue
+  );
```

Comment:
These function (especially the second one as it iterates on the form changes) should use `useMemo`.